[Android] build.gradle defaultConfig with local.properties not working properly.

Hello,

After coming back to an old flutter project and upgrading to 2.8 I found that there is a new configuration required at android/local.properties and the android/app/build.gradle is supposed to use these config variables.

However by creating a new flutter project and adding a dependency that requires minSdkVersion higher than 16, somehow, graddle doesn't recognize these properties and doesn't add them to the compiled src/debug/AndroidManifest.xml.

When I keep this configuration, it doesn't compile.

```
android/app/build.gradle:
  defaultConfig {
      // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
      applicationId "com.example.test_localconfig"
      minSdkVersion flutter.minSdkVersion
      targetSdkVersion flutter.targetSdkVersion
      versionCode flutterVersionCode.toInteger()
      versionName flutterVersionName
  }

android/local.properties
  sdk.dir=/home/tacca/Android/Sdk
  flutter.sdk=/home/tacca/tools/flutter
  flutter.buildMode=debug
  flutter.versionName=1.0.0
  flutter.versionCode=1
  flutter.minSdkVersion=21
  flutter.targetSdkVersion=30
  flutter.compileSdkVersion=30
```

However, by changing minSdkVersion and targetSdkVersion directly at build.gradle to 21 and 30, it works. 

## Steps to Reproduce

1. Create a new flutter project
2. Add a dependency that requires minSdkVersion higher than 16 (flutter_blue in my case)
3. Add the following lines to android/local.properties 

> flutter.minSdkVersion=21
> flutter.targetSdkVersion=30
> flutter.compileSdkVersion=30

4. Run the flutter project.

## Logs

<details>
<summary>flutter run -v (using flutter.minSdkVersion, not working)</summary>

```
tacca@tacca:~/Development/github/test_localconfig$ flutter run -v
[  +55 ms] executing: uname -m
[  +26 ms] Exit code 0 from: uname -m
[        ] x86_64
[   +5 ms] executing: [/home/tacca/tools/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +4 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 262b70ece1aebf84f132c51ec4cf90be605ce61b
[        ] executing: [/home/tacca/tools/flutter/] git tag --points-at 262b70ece1aebf84f132c51ec4cf90be605ce61b
[  +10 ms] Exit code 0 from: git tag --points-at 262b70ece1aebf84f132c51ec4cf90be605ce61b
[        ] 2.8.0-3.3.pre
[   +7 ms] executing: [/home/tacca/tools/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/beta
[        ] executing: [/home/tacca/tools/flutter/] git ls-remote --get-url origin
[   +3 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +64 ms] executing: [/home/tacca/tools/flutter/] git rev-parse --abbrev-ref HEAD
[   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] beta
[  +44 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +46 ms] executing: /home/tacca/Android/Sdk/platform-tools/adb devices -l
[  +31 ms] List of devices attached
           ce06160611747d2705     device usb:1-3 product:heroltexx model:SM_G930F device:herolte transport_id:2
[   +6 ms] /home/tacca/Android/Sdk/platform-tools/adb -s ce06160611747d2705 shell getprop
[  +93 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +61 ms] Skipping pub get: version match.
[  +17 ms] Found plugin flutter_blue at /home/tacca/tools/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_blue-0.8.0/
[  +89 ms] Found plugin flutter_blue at /home/tacca/tools/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_blue-0.8.0/
[  +15 ms] Generating /home/tacca/Development/github/test_localconfig/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +43 ms] ro.hardware = samsungexynos8890
[  +30 ms] Initializing file store
[   +8 ms] Skipping target: gen_localizations
[   +3 ms] gen_dart_plugin_registrant: Starting due to {InvalidatedReasonKind.inputChanged: The following inputs have updated contents:
/home/tacca/Development/github/test_localconfig/.dart_tool/package_config_subset}
[  +11 ms] Found plugin flutter_blue at /home/tacca/tools/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_blue-0.8.0/
[  +10 ms] gen_dart_plugin_registrant: Complete
[   +1 ms] Skipping target: _composite
[   +1 ms] complete
[   +4 ms] Launching lib/main.dart on SM G930F in debug mode...
[   +4 ms] /home/tacca/tools/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev /home/tacca/tools/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root
/home/tacca/tools/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata
-DFLUTTER_WEB_AUTO_DETECT=true --output-dill /tmp/flutter_tools.OFUBRI/flutter_tool.ITPGKQ/app.dill --packages
/home/tacca/Development/github/test_localconfig/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-scheme
org-dartlang-root --initialize-from-dill build/c075001b96339384a97db4862b8ab8db.cache.dill.track.dill --enable-experiment=alternative-invalidation-strategy
[   +8 ms] executing: /home/tacca/Android/Sdk/build-tools/30.0.3/aapt dump xmltree /home/tacca/Development/github/test_localconfig/build/app/outputs/flutter-apk/app.apk
AndroidManifest.xml
[   +4 ms] Exit code 0 from: /home/tacca/Android/Sdk/build-tools/30.0.3/aapt dump xmltree /home/tacca/Development/github/test_localconfig/build/app/outputs/flutter-apk/app.apk
AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1f
               A: android:compileSdkVersionCodename(0x01010573)="12" (Raw: "12")
               A: package="com.example.test_localconfig" (Raw: "com.example.test_localconfig")
               A: platformBuildVersionCode=(type 0x10)0x1f
               A: platformBuildVersionName=(type 0x10)0xc
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x15
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1e
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: uses-permission (line=15)
                 A: android:name(0x01010003)="android.permission.BLUETOOTH" (Raw: "android.permission.BLUETOOTH")
               E: uses-permission (line=16)
                 A: android:name(0x01010003)="android.permission.BLUETOOTH_ADMIN" (Raw: "android.permission.BLUETOOTH_ADMIN")
               E: uses-permission (line=17)
                 A: android:name(0x01010003)="android.permission.ACCESS_FINE_LOCATION" (Raw: "android.permission.ACCESS_FINE_LOCATION")
               E: application (line=19)
                 A: android:label(0x01010001)="test_localconfig" (Raw: "test_localconfig")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:name(0x01010003)="android.app.Application" (Raw: "android.app.Application")
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=25)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.test_localconfig.MainActivity" (Raw: "com.example.test_localconfig.MainActivity")
                   A: android:exported(0x01010010)=(type 0x12)0xffffffff
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=40)
                     A: android:name(0x01010003)="io.flutter.embedding.android.NormalTheme" (Raw: "io.flutter.embedding.android.NormalTheme")
                     A: android:resource(0x01010025)=@0x7f0a0001
                   E: intent-filter (line=44)
                     E: action (line=45)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=47)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=54)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[   +5 ms] executing: /home/tacca/Android/Sdk/platform-tools/adb -s ce06160611747d2705 shell -x logcat -v time -t 1
[  +12 ms] <- compile package:test_localconfig/main.dart
[ +134 ms] --------- beginning of main
                    12-19 19:34:40.990 E/adbd    (28387): service_to_fd: shell:logcat -v time -t 1
[   +7 ms] executing: /home/tacca/Android/Sdk/platform-tools/adb version
[   +4 ms] Android Debug Bridge version 1.0.41
           Version 31.0.2-7242960
           Installed as /home/tacca/Android/Sdk/platform-tools/adb
[   +1 ms] executing: /home/tacca/Android/Sdk/platform-tools/adb start-server
[   +5 ms] Building APK
[  +13 ms] Running Gradle task 'assembleDebug'...
[   +2 ms] Using gradle from /home/tacca/Development/github/test_localconfig/android/gradlew.
[  +30 ms] executing: /usr/local/android-studio/jre/bin/java -version
[  +80 ms] Exit code 0 from: /usr/local/android-studio/jre/bin/java -version
[        ] openjdk version "11.0.8" 2020-07-14
           OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)
           OpenJDK 64-Bit Server VM (build 11.0.8+0-b944-6842174, mixed mode)
[   +1 ms] executing: [/home/tacca/Development/github/test_localconfig/android/] /home/tacca/Development/github/test_localconfig/android/gradlew -Pverbose=true
-Ptarget-platform=android-arm64 -Ptarget=/home/tacca/Development/github/test_localconfig/lib/main.dart -Pbase-application-name=android.app.Application
-Pdart-defines=RkxVVFRFUl9XRUJfQVVUT19ERVRFQ1Q9dHJ1ZQ== -Pdart-obfuscation=false -Ptrack-widget-creation=true -Ptree-shake-icons=false -Pfilesystem-scheme=org-dartlang-root assembleDebug
[+2786 ms] > Task :app:compileFlutterBuildDebug
[        ] [  +56 ms] executing: uname -m
[        ] [  +28 ms] Exit code 0 from: uname -m
[        ] [        ] x86_64
[        ] [   +4 ms] executing: [/home/tacca/tools/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [   +4 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [        ] 262b70ece1aebf84f132c51ec4cf90be605ce61b
[        ] [        ] executing: [/home/tacca/tools/flutter/] git tag --points-at 262b70ece1aebf84f132c51ec4cf90be605ce61b
[        ] [  +10 ms] Exit code 0 from: git tag --points-at 262b70ece1aebf84f132c51ec4cf90be605ce61b
[        ] [        ] 2.8.0-3.3.pre
[        ] [   +8 ms] executing: [/home/tacca/tools/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[        ] [   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] [        ] origin/beta
[        ] [        ] executing: [/home/tacca/tools/flutter/] git ls-remote --get-url origin
[        ] [   +3 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] [        ] https://github.com/flutter/flutter.git
[        ] [  +45 ms] executing: [/home/tacca/tools/flutter/] git rev-parse --abbrev-ref HEAD
[        ] [   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] [        ] beta
[        ] [  +42 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [  +74 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] [   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[        ] [  +32 ms] Initializing file store
[        ] [  +16 ms] Skipping target: gen_localizations
[        ] [  +10 ms] gen_dart_plugin_registrant: Starting due to {InvalidatedReasonKind.inputChanged: The following inputs have updated contents:
/home/tacca/Development/github/test_localconfig/.dart_tool/package_config_subset}
[   +1 ms] [  +20 ms] Found plugin flutter_blue at /home/tacca/tools/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_blue-0.8.0/
[        ] [  +20 ms] gen_dart_plugin_registrant: Complete
[   +1 ms] [        ] kernel_snapshot: Starting due to {}
[        ] [   +8 ms] /home/tacca/tools/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev /home/tacca/tools/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot
--sdk-root /home/tacca/tools/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --target=flutter --no-print-incremental-dependencies -DFLUTTER_WEB_AUTO_DETECT=true
-Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --no-link-platform --packages
/home/tacca/Development/github/test_localconfig/.dart_tool/package_config.json --output-dill
/home/tacca/Development/github/test_localconfig/.dart_tool/flutter_build/c3b98f338088a311713415eb8cf1a73a/app.dill --depfile
/home/tacca/Development/github/test_localconfig/.dart_tool/flutter_build/c3b98f338088a311713415eb8cf1a73a/kernel_snapshot.d package:test_localconfig/main.dart
[+6386 ms] [+7740 ms] kernel_snapshot: Complete
[ +300 ms] [ +335 ms] debug_android_application: Starting due to {}
[ +299 ms] [ +282 ms] debug_android_application: Complete
[ +400 ms] [ +409 ms] Persisting file store
[        ] [   +5 ms] Done persisting file store
[        ] [   +3 ms] build succeeded.
[        ] [   +8 ms] "flutter assemble" took 8,991ms.
[ +199 ms] [ +153 ms] ensureAnalyticsSent: 150ms
[        ] [        ] Running shutdown hooks
[        ] [        ] Shutdown hooks complete
[        ] [        ] exiting with code 0
[  +99 ms] > Task :app:packLibsflutterBuildDebug UP-TO-DATE
[        ] > Task :app:preBuild UP-TO-DATE
[        ] > Task :app:preDebugBuild UP-TO-DATE
[        ] > Task :flutter_blue:preBuild UP-TO-DATE
[        ] > Task :flutter_blue:preDebugBuild UP-TO-DATE
[        ] > Task :flutter_blue:compileDebugAidl NO-SOURCE
[        ] > Task :app:compileDebugAidl NO-SOURCE
[        ] > Task :flutter_blue:packageDebugRenderscript NO-SOURCE
[        ] > Task :app:compileDebugRenderscript NO-SOURCE
[        ] > Task :app:generateDebugBuildConfig UP-TO-DATE
[        ] > Task :flutter_blue:writeDebugAarMetadata UP-TO-DATE
[  +98 ms] > Task :app:checkDebugAarMetadata UP-TO-DATE
[        ] > Task :app:cleanMergeDebugAssets
[        ] > Task :app:mergeDebugShaders UP-TO-DATE
[        ] > Task :app:compileDebugShaders NO-SOURCE
[        ] > Task :app:generateDebugAssets UP-TO-DATE
[        ] > Task :flutter_blue:mergeDebugShaders UP-TO-DATE
[        ] > Task :flutter_blue:compileDebugShaders NO-SOURCE
[        ] > Task :flutter_blue:generateDebugAssets UP-TO-DATE
[        ] > Task :flutter_blue:packageDebugAssets UP-TO-DATE
[        ] > Task :app:mergeDebugAssets
[ +100 ms] > Task :app:copyFlutterAssetsDebug
[        ] > Task :app:generateDebugResValues UP-TO-DATE
[        ] > Task :app:generateDebugResources UP-TO-DATE
[        ] > Task :flutter_blue:compileDebugRenderscript NO-SOURCE
[        ] > Task :flutter_blue:generateDebugResValues UP-TO-DATE
[        ] > Task :flutter_blue:generateDebugResources UP-TO-DATE
[        ] > Task :flutter_blue:packageDebugResources UP-TO-DATE
[  +96 ms] > Task :app:createDebugCompatibleScreenManifests
[        ] > Task :app:extractDeepLinksDebug UP-TO-DATE
[        ] > Task :flutter_blue:extractDeepLinksDebug UP-TO-DATE
[        ] > Task :flutter_blue:processDebugManifest UP-TO-DATE
[        ] > Task :app:mergeDebugResources
[        ] > Task :app:processDebugMainManifest FAILED
[        ] See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.
[        ] /home/tacca/Development/github/test_localconfig/android/app/src/debug/AndroidManifest.xml Error:
[   +1 ms]      uses-sdk:minSdkVersion 16 cannot be smaller than version 19 declared in library [:flutter_blue]
/home/tacca/Development/github/test_localconfig/build/flutter_blue/intermediates/library_manifest/debug/AndroidManifest.xml as the library might be using APIs not available in 16
[        ]      Suggestion: use a compatible library with a minSdk of at most 16,
[        ]              or increase this project's minSdk version to at least 19,
[        ]              or use tools:overrideLibrary="com.pauldemarco.flutter_blue" to force usage (may lead to runtime failures)
[        ] FAILURE: Build failed with an exception.
[        ] * What went wrong:
[        ] Execution failed for task ':app:processDebugMainManifest'.
[        ] > Manifest merger failed : uses-sdk:minSdkVersion 16 cannot be smaller than version 19 declared in library [:flutter_blue]
/home/tacca/Development/github/test_localconfig/build/flutter_blue/intermediates/library_manifest/debug/AndroidManifest.xml as the library might be using APIs not available in 16
[        ]      Suggestion: use a compatible library with a minSdk of at most 16,
[        ]              or increase this project's minSdk version to at least 19,
[        ]              or use tools:overrideLibrary="com.pauldemarco.flutter_blue" to force usage (may lead to runtime failures)
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[        ] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 10s
[        ] Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
[        ] Use '--warning-mode all' to show the individual deprecation warnings.
[        ] See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings
[        ] 20 actionable tasks: 7 executed, 13 up-to-date
[ +322 ms] Running Gradle task 'assembleDebug'... (completed in 11.2s)
[   +1 ms] 
           The plugin flutter_blue requires a higher Android SDK version.
           Fix this issue by adding the following to the file /home/tacca/Development/github/test_localconfig/android/app/build.gradle:
           android {
             defaultConfig {
               minSdkVersion 19
             }
           }
           
           
           Note that your app won't be available to users running Android SDKs below 19.
           Alternatively, try to find a version of this plugin that supports these lower versions of the Android SDK.
[   +2 ms] Exception: Gradle task assembleDebug failed with exit code 1
[   +1 ms] "flutter run" took 11,939ms.
[   +2 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:674:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1140:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:288:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:236:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:94:3)
           <asynchronous suspension>
           
           
[  +68 ms] ensureAnalyticsSent: 67ms
[        ] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```
</details>


<details>
<summary>flutter doctor -v</summary>

```
tacca@tacca:~/Development/github/test_localconfig$ flutter doctor -v
[✓] Flutter (Channel beta, 2.8.0-3.3.pre, on Ubuntu 20.04.1 LTS 5.11.0-36-generic, locale en_US.UTF-8)
    • Flutter version 2.8.0-3.3.pre at /home/tacca/tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 262b70ece1 (3 weeks ago), 2021-12-01 13:00:48 -0800
    • Engine revision 06a7363b0c
    • Dart version 2.15.0 (build 2.15.0-268.18.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/tacca/Android/Sdk
    • Platform android-31, build-tools 30.0.3
    • ANDROID_HOME = /home/tacca/Android/Sdk
    • Java binary at: /usr/local/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /usr/bin/brave-browser

[✓] Android Studio (version 4.2)
    • Android Studio at /usr/local/android-studio
    • Flutter plugin version 57.0.2
    • Dart plugin version 202.8488
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] VS Code (version 1.63.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.29.0

[✓] Connected device (2 available)
    • SM G930F (mobile) • ce06160611747d2705 • android-arm64  • Android 8.0.0 (API 26)
    • Chrome (web)      • chrome             • web-javascript • Brave Browser 93.1.29.81

• No issues found!
```

--- a/examples/api/android/app/build.gradle
+++ b/examples/api/android/app/build.gradle
@@ -29,6 +29,16 @@ apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
 
@@ -48,8 +58,8 @@ android {
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "dev.flutter.flutter_api_samples"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }

--- a/examples/flutter_view/android/app/build.gradle
+++ b/examples/flutter_view/android/app/build.gradle
@@ -28,6 +28,16 @@ if (flutterVersionName == null) {
 apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
 
@@ -38,8 +48,8 @@ android {
 
     defaultConfig {
         applicationId "io.flutter.examples.flutter_view"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }

--- a/examples/hello_world/android/app/build.gradle
+++ b/examples/hello_world/android/app/build.gradle
@@ -28,6 +28,16 @@ if (flutterVersionName == null) {
 apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
 
@@ -38,8 +48,8 @@ android {
 
     defaultConfig {
         applicationId "io.flutter.examples.hello_world"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }

--- a/examples/image_list/android/app/build.gradle
+++ b/examples/image_list/android/app/build.gradle
@@ -28,6 +28,16 @@ if (flutterVersionName == null) {
 apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
 
@@ -39,8 +49,8 @@ android {
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "com.example.image_list"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }

--- a/examples/layers/android/app/build.gradle
+++ b/examples/layers/android/app/build.gradle
@@ -28,6 +28,16 @@ if (flutterVersionName == null) {
 apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
 
@@ -38,8 +48,8 @@ android {
 
     defaultConfig {
         applicationId "io.flutter.examples.layers"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }

--- a/examples/platform_channel/android/app/build.gradle
+++ b/examples/platform_channel/android/app/build.gradle
@@ -28,6 +28,16 @@ if (flutterVersionName == null) {
 apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
 
@@ -38,8 +48,8 @@ android {
 
     defaultConfig {
         applicationId "io.flutter.examples.platform_channel"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"

--- a/examples/platform_view/android/app/build.gradle
+++ b/examples/platform_view/android/app/build.gradle
@@ -28,6 +28,16 @@ if (flutterVersionName == null) {
 apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
 
@@ -38,8 +48,8 @@ android {
 
     defaultConfig {
         applicationId "io.flutter.examples.platform_view"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }

--- a/packages/flutter_tools/README.md
+++ b/packages/flutter_tools/README.md
@@ -97,7 +97,7 @@ variable to be set. The full invocation to run everything might
 therefore look something like:
 
 ```shell
-$ FLUTTER_ROOT=~/path/to/flutter-sdk
+$ export FLUTTER_ROOT=~/path/to/flutter-sdk
 $ flutter test --concurrency 1
 ```
 

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -425,16 +425,15 @@ class FlutterPlugin implements Plugin<Project> {
             // Checks if there is a mismatch between the plugin minSdkVersion and the project minSdkVersion.
             if(pluginProject.android.defaultConfig?.minSdkVersion?.apiLevel > project.android.defaultConfig?.minSdkVersion?.apiLevel) {
                 project.logger.quiet("Warning: The plugin ${pluginName} requires minSdkVersion ${pluginProject.android.defaultConfig?.minSdkVersion?.apiLevel}.")
+                showBuildConfigurationMessage = true
                 
                 if (!resolveProperty("flutter.minSdkVersion", null)) {
                     project.logger.quiet("flutter.minSdkVersion not found on ${project.projectDir.parentFile}${File.separator}local.properties, consider adding it. Using ${project.android.defaultConfig.minSdkVersion.apiLevel} as default.")
-                    showBuildConfigurationMessage = true
                 }
 
                 // Additionally checks if the targetSdkVersion was set.
                 if (!resolveProperty("flutter.targetSdkVersion", null)) {
                     project.logger.quiet("flutter.targetSdkVersion not found on ${project.projectDir.parentFile}${File.separator}local.properties, consider adding it. Using ${project.android.defaultConfig.targetSdkVersion.apiLevel} as default.")
-                    showBuildConfigurationMessage = true
                 }
             }
             

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -139,6 +139,11 @@ class FlutterPlugin implements Plugin<Project> {
     private Properties localProperties
     private String engineVersion
 
+    /**
+     * Flutter Docs Website references URLs for help messages.
+     */
+    private final String kWebsiteDeploymentAndroidBuildConfig = 'https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration'
+
     @Override
     void apply(Project project) {
         this.project = project
@@ -300,7 +305,7 @@ class FlutterPlugin implements Plugin<Project> {
             localEngine = engineOut.name
             localEngineSrcPath = engineOut.parentFile.parent
         }
-        project.android.buildTypes.all this.&addFlutterDependencies
+        project.android.buildTypes.all this.&addFlutterDependencies        
     }
 
     private static Boolean shouldShrinkResources(Project project) {
@@ -409,9 +414,34 @@ class FlutterPlugin implements Plugin<Project> {
 
         // Wait until the Android plugin loaded.
         pluginProject.afterEvaluate {
+            Boolean showBuildConfigurationMessage = false
+            
+            // Checks if there is a mismatch between the plugin compileSdkVersion and the project compileSdkVersion.
             if (pluginProject.android.compileSdkVersion > project.android.compileSdkVersion) {
                 project.logger.quiet("Warning: The plugin ${pluginName} requires Android SDK version ${pluginProject.android.compileSdkVersion.substring(8)}.")
+                showBuildConfigurationMessage = true
             }
+
+            // Checks if there is a mismatch between the plugin minSdkVersion and the project minSdkVersion.
+            if(pluginProject.android.defaultConfig?.minSdkVersion?.apiLevel > project.android.defaultConfig?.minSdkVersion?.apiLevel) {
+                project.logger.quiet("Warning: The plugin ${pluginName} requires minSdkVersion ${pluginProject.android.defaultConfig?.minSdkVersion?.apiLevel}.")
+                
+                if (!resolveProperty("flutter.minSdkVersion", null)) {
+                    project.logger.quiet("flutter.minSdkVersion not found on ${project.projectDir.parentFile}${File.separator}local.properties, consider adding it. Using ${project.android.defaultConfig.minSdkVersion.apiLevel} as default.")
+                    showBuildConfigurationMessage = true
+                }
+
+                // Additionally checks if the targetSdkVersion was set.
+                if (!resolveProperty("flutter.targetSdkVersion", null)) {
+                    project.logger.quiet("flutter.targetSdkVersion not found on ${project.projectDir.parentFile}${File.separator}local.properties, consider adding it. Using ${project.android.defaultConfig.targetSdkVersion.apiLevel} as default.")
+                    showBuildConfigurationMessage = true
+                }
+            }
+            
+            if (showBuildConfigurationMessage) {
+                project.logger.quiet("For more information about build configuration, see $kWebsiteDeploymentAndroidBuildConfig.")
+            }
+
             project.android.buildTypes.all addEmbeddingDependencyToPlugin
         }
     }

--- a/packages/flutter_tools/lib/src/android/gradle_errors.dart
+++ b/packages/flutter_tools/lib/src/android/gradle_errors.dart
@@ -390,27 +390,24 @@ final GradleHandledError minSdkVersion = GradleHandledError(
     required bool usesAndroidX,
     required bool multidexEnabled,
   }) async {
-    final File gradleFile = project.directory
+    final File localPropertiesFile = project.directory
         .childDirectory('android')
-        .childDirectory('app')
-        .childFile('build.gradle');
+        .childFile('local.properties');
 
     final Match? minSdkVersionMatch = _minSdkVersionPattern.firstMatch(line);
     assert(minSdkVersionMatch?.groupCount == 3);
 
     final String textInBold = globals.logger.terminal.bolden(
-      'Fix this issue by adding the following to the file ${gradleFile.path}:\n'
-      'android {\n'
-      '  defaultConfig {\n'
-      '    minSdkVersion ${minSdkVersionMatch?.group(2)}\n'
-      '  }\n'
-      '}\n'
+      'Fix this issue by adding the following to the file ${localPropertiesFile.path}:\n'
+      '\n'
+      'flutter.minSdkVersion=${minSdkVersionMatch?.group(2)}\n'
     );
     globals.printBox(
       'The plugin ${minSdkVersionMatch?.group(3)} requires a higher Android SDK version.\n'
       '$textInBold\n'
       "Note that your app won't be available to users running Android SDKs below ${minSdkVersionMatch?.group(2)}.\n"
-      'Alternatively, try to find a version of this plugin that supports these lower versions of the Android SDK.',
+      'Alternatively, try to find a version of this plugin that supports these lower versions of the Android SDK.\n'
+      'For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration',
       title: _boxTitle,
     );
     return GradleBuildStatus.exit;
@@ -518,8 +515,8 @@ final GradleHandledError minCompileSdkVersionHandler = GradleHandledError(
     required bool usesAndroidX,
     required bool multidexEnabled,
   }) async {
-    final Match? minSdkVersionMatch = _minCompileSdkVersionPattern.firstMatch(line);
-    assert(minSdkVersionMatch?.groupCount == 1);
+    final Match? minCompileSdkVersionMatch = _minCompileSdkVersionPattern.firstMatch(line);
+    assert(minCompileSdkVersionMatch?.groupCount == 1);
 
     final File gradleFile = project.directory
         .childDirectory('android')
@@ -529,7 +526,7 @@ final GradleHandledError minCompileSdkVersionHandler = GradleHandledError(
       '${globals.logger.terminal.warningMark} Your project requires a higher compileSdkVersion.\n'
       'Fix this issue by bumping the compileSdkVersion in ${gradleFile.path}:\n'
       'android {\n'
-      '  compileSdkVersion ${minSdkVersionMatch?.group(1)}\n'
+      '  compileSdkVersion ${minCompileSdkVersionMatch?.group(1)}\n'
       '}',
       title: _boxTitle,
     );

--- a/packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl
+++ b/packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl
@@ -25,6 +25,18 @@ apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+    logger.quiet("flutter.minSdkVersion not found on android/local.properties, consider adding it. Using ${flutterMinSdkVersion} as default.")
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+    logger.quiet("flutter.targetSdkVersion not found on android/local.properties, consider adding it. Using ${flutterTargetSdkVersion} as default.")
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
     ndkVersion flutter.ndkVersion
@@ -37,8 +49,8 @@ android {
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "{{androidIdentifier}}"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }

--- a/packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl
+++ b/packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl
@@ -28,13 +28,11 @@ apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
 if (flutterMinSdkVersion == null) {
     flutterMinSdkVersion = flutter.minSdkVersion
-    logger.quiet("flutter.minSdkVersion not found on android/local.properties, consider adding it. Using ${flutterMinSdkVersion} as default.")
 }
 
 def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
 if (flutterTargetSdkVersion == null) {
     flutterTargetSdkVersion = flutter.targetSdkVersion
-    logger.quiet("flutter.targetSdkVersion not found on android/local.properties, consider adding it. Using ${flutterTargetSdkVersion} as default.")
 }
 
 android {

--- a/packages/flutter_tools/templates/app_shared/android-kotlin.tmpl/app/build.gradle.tmpl
+++ b/packages/flutter_tools/templates/app_shared/android-kotlin.tmpl/app/build.gradle.tmpl
@@ -25,6 +25,18 @@ apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+    logger.quiet("flutter.minSdkVersion not found on android/local.properties, consider adding it. Using ${flutterMinSdkVersion} as default.")
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+    logger.quiet("flutter.targetSdkVersion not found on android/local.properties, consider adding it. Using ${flutterTargetSdkVersion} as default.")
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
     ndkVersion flutter.ndkVersion
@@ -45,8 +57,8 @@ android {
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "{{androidIdentifier}}"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }

--- a/packages/flutter_tools/templates/app_shared/android-kotlin.tmpl/app/build.gradle.tmpl
+++ b/packages/flutter_tools/templates/app_shared/android-kotlin.tmpl/app/build.gradle.tmpl
@@ -28,13 +28,11 @@ apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
 if (flutterMinSdkVersion == null) {
     flutterMinSdkVersion = flutter.minSdkVersion
-    logger.quiet("flutter.minSdkVersion not found on android/local.properties, consider adding it. Using ${flutterMinSdkVersion} as default.")
 }
 
 def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
 if (flutterTargetSdkVersion == null) {
     flutterTargetSdkVersion = flutter.targetSdkVersion
-    logger.quiet("flutter.targetSdkVersion not found on android/local.properties, consider adding it. Using ${flutterTargetSdkVersion} as default.")
 }
 
 android {

--- a/packages/flutter_tools/templates/module/android/library_new_embedding/Flutter.tmpl/build.gradle.tmpl
+++ b/packages/flutter_tools/templates/module/android/library_new_embedding/Flutter.tmpl/build.gradle.tmpl
@@ -26,6 +26,18 @@ if (flutterVersionName == null) {
 apply plugin: 'com.android.library'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+    logger.quiet("flutter.minSdkVersion not found on android/local.properties, consider adding it. Using ${flutterMinSdkVersion} as default.")
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+    logger.quiet("flutter.targetSdkVersion not found on android/local.properties, consider adding it. Using ${flutterTargetSdkVersion} as default.")
+}
+
 group '{{androidIdentifier}}'
 version '1.0'
 
@@ -39,8 +51,8 @@ android {
     }
 
     defaultConfig {
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }

--- a/packages/flutter_tools/templates/module/android/library_new_embedding/Flutter.tmpl/build.gradle.tmpl
+++ b/packages/flutter_tools/templates/module/android/library_new_embedding/Flutter.tmpl/build.gradle.tmpl
@@ -29,13 +29,11 @@ apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
 if (flutterMinSdkVersion == null) {
     flutterMinSdkVersion = flutter.minSdkVersion
-    logger.quiet("flutter.minSdkVersion not found on android/local.properties, consider adding it. Using ${flutterMinSdkVersion} as default.")
 }
 
 def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
 if (flutterTargetSdkVersion == null) {
     flutterTargetSdkVersion = flutter.targetSdkVersion
-    logger.quiet("flutter.targetSdkVersion not found on android/local.properties, consider adding it. Using ${flutterTargetSdkVersion} as default.")
 }
 
 group '{{androidIdentifier}}'

--- a/packages/flutter_tools/test/commands.shard/permeable/create_test.dart
+++ b/packages/flutter_tools/test/commands.shard/permeable/create_test.dart
@@ -2479,7 +2479,7 @@ void main() {
     expect(env['flutter'].allows(Version(2, 4, 9)), false);
   });
 
-  testUsingContext('default app uses flutter default versions', () async {
+  testUsingContext('default app uses local properties or flutter default versions', () async {
     Cache.flutterRoot = '../..';
 
     final CreateCommand command = CreateCommand();
@@ -2493,7 +2493,8 @@ void main() {
 
     expect(buildContent.contains('compileSdkVersion flutter.compileSdkVersion'), true);
     expect(buildContent.contains('ndkVersion flutter.ndkVersion'), true);
-    expect(buildContent.contains('targetSdkVersion flutter.targetSdkVersion'), true);
+    expect(buildContent.contains('minSdkVersion flutterMinSdkVersion'), true);
+    expect(buildContent.contains('targetSdkVersion flutterTargetSdkVersion'), true);
   });
 
   testUsingContext('Linux plugins handle partially camel-case project names correctly', () async {

--- a/packages/flutter_tools/test/general.shard/android/gradle_errors_test.dart
+++ b/packages/flutter_tools/test/general.shard/android/gradle_errors_test.dart
@@ -778,16 +778,15 @@ assembleProfile
           '\n'
           '┌─ Flutter Fix ─────────────────────────────────────────────────────────────────────────────────┐\n'
           '│ The plugin webview_flutter requires a higher Android SDK version.                             │\n'
-          '│ Fix this issue by adding the following to the file /android/app/build.gradle:                 │\n'
-          '│ android {                                                                                     │\n'
-          '│   defaultConfig {                                                                             │\n'
-          '│     minSdkVersion 19                                                                          │\n'
-          '│   }                                                                                           │\n'
-          '│ }                                                                                             │\n'
+          '│ Fix this issue by adding the following to the file /android/local.properties:                 │\n'
+          '│                                                                                               │\n'
+          '│ flutter.minSdkVersion=19                                                                      │\n'
           '│                                                                                               │\n'
           "│ Note that your app won't be available to users running Android SDKs below 19.                 │\n"
           '│ Alternatively, try to find a version of this plugin that supports these lower versions of the │\n'
           '│ Android SDK.                                                                                  │\n'
+          '│ For more information, see:                                                                    │\n'
+          '│ https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration                 │\n'
           '└───────────────────────────────────────────────────────────────────────────────────────────────┘\n'
         )
       );

--- a/packages/integration_test/example/android/app/build.gradle
+++ b/packages/integration_test/example/android/app/build.gradle
@@ -28,6 +28,16 @@ if (flutterVersionName == null) {
 apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
+def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
+if (flutterMinSdkVersion == null) {
+    flutterMinSdkVersion = flutter.minSdkVersion
+}
+
+def flutterTargetSdkVersion = localProperties.getProperty('flutter.targetSdkVersion')
+if (flutterTargetSdkVersion == null) {
+    flutterTargetSdkVersion = flutter.targetSdkVersion
+}
+
 android {
     compileSdkVersion flutter.compileSdkVersion
 
@@ -39,8 +49,8 @@ android {
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "com.example.integration_test_example"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion flutter.targetSdkVersion
+        minSdkVersion flutterMinSdkVersion
+        targetSdkVersion flutterTargetSdkVersion
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"

